vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/NoRecordIndexQuery.java

```diff
@@ -0,0 +1,22 @@
+package mil.army.usace.hec.vortex.temporal;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+
+class NoRecordIndexQuery implements RecordIndexQuery {
+    @Override
+    public List<Integer> query(ZonedDateTime startTime, ZonedDateTime endTime) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public ZonedDateTime getEarliestStartTime() {
+        return null;
```

Comment:
Just make sure the client code can handle `null`.

Reply:
Updated `TemporalDataReader` public API getters to return `Optional` where the return value may be null